opentelemetry/benches/attributes.rs

```diff
@@ -10,7 +10,7 @@ RAM: 64.0 GB
 | CreateTupleKeyValue                              |      671 ps |
 | CreateOtelKeyValueArray                          |     18.4 ns |
 | CreateOtelKeyValueArrayWithMixedValueTypes       |     18.1 ns |
-| CreateOtelKeyValueArrayWithNonStaticValues       |     90.1 ns |
+| CreateOtelKeyValueArrayWithNonStaticValues       |     61.2 ns |
```

Comment:
@cijothomas The changes in this PR are helpful for this scenario.

Reply:
Can you add metric e2e benchmarks too? the counter.add and friends